Add media information to export command (#15107)
matrix-org/synapse: docs/usage/administration/admin_faq.md

│       ├───state
│       ├───invite_state
│       └───knock_state
├───user_data
│   ├───account_data
│   │   ├───global
│   │   └───<room_id>
│   ├───connections
│   ├───devices
│   └───profile
└───media_ids
    └───<media_id>
```

> The `media_ids` folder contains only the metadata of the media, not the media itself.

Reviewer: I think this needs to say that it is the data the user uploaded. (Which I'm not even sure is correct -- is that what this is?)

Author: Yes, this is from media uploaded by this user.

> The `media_ids` folder contains only the metadata of the media, not the media itself.
> Furthermore, only the `media_ids` that Synapse manages itself are exported.
> If another media repository (eg. [matrix-media-repo](https://github.com/turt2live/matrix-media-repo))
> is used, the data must be exported at this one.

Reviewer: ```suggestion
The `media_ids` folder contains only the metadata of the media, not the media itself.
Furthermore, only the `media_ids` that Synapse manages itself are exported.
If another media repository (e.g. [matrix-media-repo](https://github.com/turt2live/matrix-media-repo))
is used, the data must be exported separately.
```

is used, the data must be exported separately.

With the `media_ids` the media files can be downloaded.
Media that have been sent in encrypted rooms are only retrieved in encrypted form.
The following script can help with download the media files:

```bash
#!/usr/bin/env bash

# Parameters
#
#   source_directory: Directory which contains the export with the media_ids.
#   target_directory: Directory into which all files are to be downloaded.
#   repository_url: Address of the media repository resp. media worker.
#   serverName: Name of the server (`server_name` from homeserver.yaml).
#
#   Example:
#       ./download_media.sh /tmp/export_data/media_ids/ /tmp/export_data/media_files/ http://localhost:8008 matrix.example.com

source_directory=$1
target_directory=$2
repository_url=$3
serverName=$4

mkdir -p $target_directory

for file in $source_directory/*; do
    filename=$(basename ${file})
    url=$repository_url/_matrix/media/v3/download/$serverName/$filename
    echo "Downloading $filename - $url"
    if ! wget -o /dev/null -P $target_directory $url; then
        echo "Could not download $filename"
    fi
done

> ```

Reviewer: Sounds like this might make another good command! 👍
